vpr/src/place/placer.cpp

```diff
@@ -371,14 +371,22 @@ void Placer::place() {
     log_printer_.print_post_placement_stats();
 }
 
-void Placer::copy_locs_to_global_state(PlacementContext& place_ctx) {
+void Placer::copy_locs_to_global_state() {
+    auto& mutable_palce_ctx = g_vpr_ctx.mutable_placement();
+    auto& mutable_routing_ctx = g_vpr_ctx.mutable_routing();
+
     // the placement location variables should be unlocked before being accessed
-    place_ctx.unlock_loc_vars();
+    mutable_palce_ctx.unlock_loc_vars();
 
     // copy the local location variables into the global state
-    auto& global_blk_loc_registry = place_ctx.mutable_blk_loc_registry();
+    auto& global_blk_loc_registry = mutable_palce_ctx.mutable_blk_loc_registry();
     global_blk_loc_registry = placer_state_.blk_loc_registry();
 
+    auto [chanx_util, chany_util] = net_cost_handler_.estimate_routing_chann_util();
```

Comment:
Not sure we should have a side effect like this in a routine ...suggest instead directly calling a routine to do this at the end of placement.  Or you could rename it but still not sure that's as good.

Reply:
At least rename and update the comment of this routine.